src/Mvc/Mvc.Core/src/ConsumesAttribute.cs

```diff
@@ -64,6 +91,14 @@ public ConsumesAttribute(string contentType, params string[] otherContentTypes)
         /// </summary>
         public MediaTypeCollection ContentTypes { get; set; }
 
+        internal readonly Type? _requestType;
+
+        internal readonly List<string> _contentTypes = new();
```

Comment:
I think you can avoid exposing fields as internal types. If you need this for tests, you can always downcast to the interface:

e.g.

```C#
var consumes = new ConsumesAttribute(...);
var acceptMetadata = (IAcceptMetadata)consumes;

Assert.Equal(typeof(Foo), acceptMetadata.RequestType);
```

We generally don't expose fields outside of a type.